src/main/gov/nasa/jpf/psyco/search/util/region/RegionUtil.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (C) 2015, United States Government, as represented by the 
+ * Administrator of the National Aeronautics and Space Administration.
+ * All rights reserved.
+ *
+ * The PSYCO: A Predicate-based Symbolic Compositional Reasoning environment 
+ * platform is licensed under the Apache License, Version 2.0 (the "License"); you 
+ * may not use this file except in compliance with the License. You may obtain a 
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0. 
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed 
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR 
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the 
+ * specific language governing permissions and limitations under the License.
+ */
+package gov.nasa.jpf.psyco.search.util.region;
+
+import gov.nasa.jpf.constraints.api.ConstraintSolver;
+import gov.nasa.jpf.constraints.api.Expression;
+import gov.nasa.jpf.constraints.api.Valuation;
+import gov.nasa.jpf.constraints.api.ValuationEntry;
+import gov.nasa.jpf.constraints.api.Variable;
+import gov.nasa.jpf.constraints.expressions.Negation;
+import gov.nasa.jpf.constraints.expressions.Quantifier;
+import gov.nasa.jpf.constraints.expressions.QuantifierExpression;
+import gov.nasa.jpf.constraints.util.ExpressionUtil;
+import gov.nasa.jpf.psyco.search.datastructures.region.Region;
+import gov.nasa.jpf.psyco.search.datastructures.state.State;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Logger;
+/**
+ * Supporting class to performe a operations on Regions.
+ * Supported operations are renamin, exists, conjunction, difference.
+ * It is used in the search algorithms. Renaming must be implemented in a 
+ * subclass so that it fits the state entries.
+ */
+public abstract class RegionUtil<V extends State<?>, T extends Region<?, V>> {
+
+  private long unique;
+  ConstraintSolver solver;
+  protected final Logger logger;
+
+  public RegionUtil(ConstraintSolver solver) {
+    this.solver = solver;
+    this.unique = 1L;
+    logger = Logger.getLogger("psyco");
+  }
+
+  public T disjunction(T regionA, T regionB) {
+    T disjunctedRegion = (T) regionA.createNewRegion();
+    disjunctedRegion.putAll(regionA);
+    for (String key : regionB.keySet()) {
+      //This assumes, that state names are unique to work!
+      if (disjunctedRegion.containsKey(key)) {
```

Comment:
It is okay, as long as the state identifier are unique which is guaranteed by the search region construction.